MediaProcessor/tests/FFmpegControllerTester.cpp

```diff
@@ -0,0 +1,69 @@
+#include <gtest/gtest.h>
+
+#include <filesystem>
+#include <nlohmann/json.hpp>
+
+#include "../src/FFmpegController.h"
+#include "TestUtils.h"
+
+namespace fs = std::filesystem;
+namespace MediaProcessor::Tests {
+
+fs::path testMediaPath = TEST_MEDIA_DIR;
+
+class FFmpegControllerTester : public ::testing::Test {
+   protected:
+    fs::path testVideoPath;
+    fs::path testAudioProcessedPath;
+    fs::path testOutputDir;
+    TestUtils::TestConfigFile testConfigFile;
+    FFmpegConfigManager configManager;
+
+    FFmpegControllerTester() : configManager(FFmpegConfigManager()) {}
+
+    void assertFileExists(fs::path path) {
+        ASSERT_TRUE(fs::exists(path)) << path << " not found.";
+    }
+
+    void SetUp() override {
+        fs::path currentPath = fs::current_path();
+
+        testVideoPath = testMediaPath / "test_video.mkv";
+        testAudioProcessedPath = testMediaPath / "test_audio_processed.wav";
+
+        assertFileExists(testVideoPath);
+        assertFileExists(testAudioProcessedPath);
+
+        testOutputDir = currentPath / "test_output";
+        fs::create_directories(testOutputDir);
+
+        testConfigFile.changeConfigOptions("use_thread_cap", true, "max_threads_if_capped", 1);
+    }
+
+    void TearDown() override {
+        fs::remove_all(testOutputDir);
+    }
+};
+
+// ClassName_MethodName_StateUnderTest_ExpectedBehavior gtest std naming convention
+TEST_F(FFmpegControllerTester, ExtractAudio_ExtractsAudioCorrectly) {
+    fs::path testAudioOutputPath = testOutputDir / "test_output_audio.wav";
+
+    configManager.setOverwrite(true);
+    configManager.setInputFilePath(testVideoPath);
+    configManager.setAudioSampleRate(48000);
+    configManager.setAudioChannels(1);
+    configManager.setAudioCodec(AudioCodec::AAC);
+    configManager.setOutputFilePath(testAudioOutputPath);
+
+    FFmpegController controller = FFmpegController(configManager);
+
+    EXPECT_EQ(controller.extractAudio(), true);
+
+    EXPECT_TRUE(fs::exists(testAudioOutputPath));
+
+    EXPECT_TRUE(
+        TestUtils::CompareFiles::compareAudioFiles(testAudioOutputPath, testAudioProcessedPath));
+}
```

Comment:
I haven't run this yet, but looks reasonable. Could you also add tests for the remaining public member fns and I'll run the tests on my end as well.

Reply:
Hi @omeryusufyagci ...just so you know I'm working on this ;)

I needed to do some refactoring. When implementing the controller I added generateChunkFile as a public method, but that presents problems for the tests because the helpers it are private to AudioController. To me it seems easier for the user if we add splitAudioIntoChunks as public to the controller, with helpers like generateChunkFile as private.

Short version: I'm making my life harder in the short term by moving more pieces out of AudioController so we can hopefully have an easier time moving forward with tests and setting up a nice API :)

Reply:
Thanks @andcscott, that sounds good! 

Some ffmpeg commands work both for audio and video, perhaps this works the same as well. If that's the case, we can name it like split**Media**...and maybe a name like `chunkMedia()` will read easier for the public fn as well.